src/cdk/table/table.ts

```diff
@@ -641,9 +647,13 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
       });
   }
 
+  ngAfterContentInit() {
+    this._hasInitialized = true;
+  }
+
   ngAfterContentChecked() {
     // Only start re-rendering in `ngAfterContentChecked` after the first render.
-    if (this._hasRendered) {
+    if (this._canRender()) {
```

Comment:
If calls to `_render` will always need to be wrapped with a conditional, should we just add this check to the beginning of `_render` and document that it will no-op if the table cannot render

Reply:
I wanted to avoid cases where `_render` fails silently since it'll be much harder to track down.